sklearn/model_selection/_split.py

```diff
@@ -1727,6 +1729,144 @@ def split(self, X, y, groups=None):
         return super().split(X, y, groups)
 
 
+class StratifiedGroupShuffleSplit(StratifiedShuffleSplit):
+    """Stratified GroupShuffleSplit cross-validator
+
+    Provides randomized train/test indices to split data according to a
+    third-party provided group. This group information can be used to encode
+    arbitrary domain specific stratifications of the samples as integers.
+
+    This cross-validation object is a merge of GroupShuffleSplit and
+    StratifiedShuffleSplit, which returns stratified randomized folds. The
+    folds are made by preserving the percentage of groups for each class.
+
+    Note: like the StratifiedShuffleSplit strategy, stratified random group
+    splits do not guarantee that all folds will be different, although this is
```

Comment:
I don't understand this sentence. What do you mean by "folds not being different"?

Reply:
That text is copied from `StratifiedShuffleSplit` and the meaning behind it is that shuffle splitting does not guarantee that each split will be different than all any of the others.

Reply:
Ah. I think in sklearn we call folds the partitions in the split, not the repetitions.

Reply:
I think partitions/splits/folds that's what is meant here and I believe in `StratifiedShuffleSplit`.  With randomized splits there is no guarantee that a partition will be different than other ones.